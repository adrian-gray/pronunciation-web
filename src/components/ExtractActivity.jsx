import React from 'react'

import CommonWords from './../components/CommonWords'
import Dialogues from './../components/Dialogues'
import FindTheWords from './../components/FindTheWords'
import FourInARow from './../components/FourInARow'
import HowToPronounce from './../components/HowToPronounce'
import NewsStories from './../components/NewsStories'
import MinimalPairs from './../components/MinimalPairs'
import Movie from './../components/Movie'
import Pending from './../components/Pending'
import Phrases from './../components/Phrases'
import SayTheSentences from './../components/SayTheSentences'
import TongueTwisters from './../components/TongueTwisters'
import Words from './../components/Words'
import WordsMaze from './../components/WordsMaze'

const ExtractActivity = (params) => {
  const { activityName, data, subscriptionLevel } = params

  let activityComponent = null
  const activity = data.activities[activityName]
  const requiredSubscriptionLevel = activity && activity['subscription']
  const userAuth = subscriptionLevel >= requiredSubscriptionLevel

  switch (activityName) {
    case 'common words':
      activityComponent = (
        <CommonWords
          ipa={data.ipa}
          words={activity.words}
          userAuth={userAuth}
        />
      )
      break

    case 'dialogues':
      activityComponent = (
        <Dialogues
          ipa={data.ipa}
          dialogues={activity.dialogues}
          userAuth={userAuth}
        />
      )
      break

    case 'find the words':
      activityComponent = (
        <FindTheWords
          ipa={data.ipa}
          tag={data.tag}
          titleStart='Select the words with an '
          titleEnd='sound'
          words={activity.words}
          correct={activity.correct}
          userAuth={userAuth}
        />
      )
      break

    case 'odd one out':
      activityComponent = (
        <FindTheWords
          ipa={data.ipa}
          titleStart="Select the words that DON'T have an "
          titleEnd='sound'
          words={activity.words}
          correct={activity.correct}
          userAuth={userAuth}
        />
      )
      break

    case 'four in a row':
      activityComponent = (
        <FourInARow
          correct={activity.correct}
          example={activity.example}
          exampleHilite={activity.exampleHilite}
          ipa={data.ipa}
          rows={activity.rows}
          tag={data.tag}
          userAuth={userAuth}
        />
      )
      break

    case 'how to pronounce':
      activityComponent = (
        <HowToPronounce
          ipa={data.ipa}
          arr={activity.text}
          image={activity.image}
          userAuth={userAuth}
        />
      )
      break

    case 'minimal pairs':
      activityComponent = (
        <MinimalPairs
          ipa={data.ipa}
          tag={data.tag}
          pairs={activity.pairs}
          userAuth={userAuth}
        />
      )
      break

    case 'movie':
      activityComponent = (
        <Movie
          ipa={data.ipa}
          url={activity.url}
          userAuth={userAuth}
        />
      )
      break

    case 'new stories': case 'beginner news stories': case 'intermediate news stories':
      activityComponent = (
        <NewsStories
          title={activity.title}
          headline={activity.headline}
          options={activity.options}
          sentences={activity.sentences}
          answers={activity.answers}
          userAuth={userAuth}
        />
      )
      break

    case 'phrases':
      activityComponent = (
        <Phrases
          phrases={activity.phrases}
          ipa={data.ipa}
          userAuth={userAuth}
        />
      )
      break

    case 'say the sentences':
      activityComponent = (
        <SayTheSentences
          ipa={data.ipa}
          sentences={activity.sentences}
        />
      )
      break

    case 'tongue twisters':
      activityComponent = (
        <TongueTwisters
          tag={data.tag}
          tongueTwisters={activity.twisters}
          userAuth={userAuth}
        />
      )
      break

    case 'words':
      activityComponent = (
        <Words
          ipa={data.ipa}
          words={activity.words}
          userAuth={userAuth}
        />
      )
      break

    case 'words maze':
      activityComponent = (
        <WordsMaze
          tag={data.tag}
          words={activity.words}
          correct={activity.correct}
          userAuth={userAuth}
        />
      )
      break

    case 'sort the words':
    case 'hear the words':
    case 'pronunciation journey':
    case 'rhyming words':
    case 'count the sounds':
    case 'write the minimal pair':
    case 'join to make words':
    case 'listen to the sounds':
    case 'minimal pairs dominos':
      activityComponent = <Pending name={activityName} />
      break

    case undefined:
      break

    default:
      activityComponent = undefined
  }

  return activityComponent
}

export default ExtractActivity
