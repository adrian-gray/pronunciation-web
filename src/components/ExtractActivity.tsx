import React from "react";

import CommonWords from "./../activities/CommonWords";
import Dialogues from "./../activities/Dialogues";
import FindTheWords from "./../activities/FindTheWords";
import OddOneOut from "./../activities/OddOneOut";
import HowToPronounce from "./../activities/HowToPronounce";
import NewsStories from "./../activities/NewsStories";
import MinimalPairs from "./../activities/MinimalPairs";
import MissingSounds from "./../activities/MissingSounds";
import Movie from "./../activities/Movie";
import Pending from "./../activities/Pending";
import Phrases from "./../activities/Phrases";
import SayTheSentences from "./../activities/SayTheSentences";
import TongueTwisters from "./../activities/TongueTwisters";
import Words from "./../activities/Words";
import WordsMaze from "./../activities/WordsMaze";

const ExtractActivity = (params: {
  props: any;
  activityName: string;
  data: any;
  subscriptionLevel: number;
}) => {
  const { props, activityName, data, subscriptionLevel } = params;
  let activityComponent = null;
  const activity = data.activities[activityName];
  const requiredSubscriptionLevel: number =
    (activity && activity["subscription"]) || 0;
  const isUserAuth: boolean = subscriptionLevel >= requiredSubscriptionLevel;

  switch (activityName) {
    case "common words":
      activityComponent = (
        <CommonWords
          ipa={data.ipa}
          words={activity.words}
          isUserAuth={isUserAuth}
        />
      );
      break;

    case "dialogues":
      activityComponent = (
        <Dialogues
          ipa={data.ipa}
          dialogues={activity.dialogues}
          isUserAuth={isUserAuth}
        />
      );
      break;

    case "find the words":
      activityComponent = (
        <FindTheWords
          ipa={data.ipa}
          tag={data.tag}
          words={activity.words}
          correct={activity.correct}
          isHearTheDifference={false}
          isUserAuth={isUserAuth}
        />
      );
      break;

    case "missing sounds":
      activityComponent = (
        <MissingSounds
          ipa={data.ipa}
          tag={data.tag}
          words={activity.words}
          answers={activity.answers}
          isHearTheDifference={false}
          isUserAuth={isUserAuth}
        />
      );
      break;

    case "odd one out1":
      activityComponent = (
        <FindTheWords
          ipa={data.ipa}
          tag={data.tag}
          words={activity.words}
          correct={activity.correct}
          isUserAuth={isUserAuth}
          isHearTheDifference={false}
        />
      );
      break;

    case "odd one out":
      activityComponent = (
        <OddOneOut
          correct={activity.correct}
          example={activity.example}
          exampleHilite={activity.exampleHilite}
          ipa={data.ipa}
          tag={data.tag}
          rows={activity.rows}
          isUserAuth={isUserAuth}
        />
      );
      break;

    case "how to pronounce":
      activityComponent = (
        <HowToPronounce
          ipa={data.ipa}
          arr={activity.text}
          image={activity.image}
        />
      );
      break;

    case "minimal pairs":
      activityComponent = (
        <MinimalPairs
          ipa={data.ipa}
          tag={data.tag}
          pairs={activity.pairs}
          isUserAuth={isUserAuth}
        />
      );
      break;

    case "movie":
      activityComponent = (
        <Movie ipa={data.ipa} url={activity.url} isUserAuth={isUserAuth} />
      );
      break;

    case "new stories":
    case "beginner news stories":
    case "intermediate news stories":
      activityComponent = (
        <NewsStories
          level={activity.level}
          title={activity.title}
          headline={activity.headline}
          options={activity.options}
          sentences={activity.sentences}
          answers={activity.answers}
          isUserAuth={isUserAuth}
          {...props}
        />
      );
      break;

    case "phrases":
      activityComponent = <Phrases phrases={activity.phrases} ipa={data.ipa} />;
      break;

    case "say the sentences":
      activityComponent = (
        <SayTheSentences ipa={data.ipa} sentences={activity.sentences} />
      );
      break;

    case "tongue twisters":
      activityComponent = (
        <TongueTwisters
          tag={data.tag}
          tongueTwisters={activity.twisters}
          isUserAuth={isUserAuth}
        />
      );
      break;

    case "words":
      activityComponent = <Words ipa={data.ipa} words={activity.words} />;
      break;

    case "words maze":
      activityComponent = (
        <WordsMaze
          ipa={data.ipa}
          tag={data.tag}
          words={activity.words}
          correct={activity.correct}
          isUserAuth={isUserAuth}
        />
      );
      break;

    case "sort the words":
    case "hear the words":
    case "pronunciation journey":
    case "rhyming words":
    case "count the sounds":
    case "write the minimal pair":
    case "join to make words":
    case "listen to the sounds":
    case "minimal pairs dominos":
      activityComponent = <Pending name={activityName} />;
      break;

    default:
      activityComponent = null;
  }

  return activityComponent;
};

export default ExtractActivity;
