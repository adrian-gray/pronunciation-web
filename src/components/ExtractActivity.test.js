import React from "react";
import { render, cleanup } from "@testing-library/react";

import ExtractActivity from "./ExtractActivity";

afterEach(cleanup);

test("ExtractActivity", async () => {
  const props = {
    props: { poop: "smelly" },
    data: {
      ipa: "a:",
      tag: "mitch",
      activities: {
        "common words": {
          words: [
            "foo",
            "bar",
            "baz",
            "quox",
            "apple",
            "banana",
            "chad",
            "elle"
          ]
        }
      },
      subscriptionLevel: 1
    },
    activityName: "common words"
  };

  const { container } = render(<ExtractActivity {...props} />);

  expect(container).toMatchSnapshot();
});
