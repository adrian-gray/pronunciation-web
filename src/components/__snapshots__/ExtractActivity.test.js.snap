// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtractActivity 1`] = `
<div>
  <div
    class="makeStyles-headspace-3"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      Common words with 
      <span>
        a:
      </span>
       
    </h5>
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        class="MuiTable-root"
      >
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root makeStyles-cell-1 MuiTableCell-body"
            >
              <p
                class="MuiTypography-root makeStyles-largeText-2 MuiTypography-body1"
              >
                <span>
                  foo
                </span>
                 
              </p>
            </td>
            <td
              class="MuiTableCell-root makeStyles-cell-1 MuiTableCell-body"
            >
              <p
                class="MuiTypography-root makeStyles-largeText-2 MuiTypography-body1"
              >
                <span>
                  bar
                </span>
                 
              </p>
            </td>
            <td
              class="MuiTableCell-root makeStyles-cell-1 MuiTableCell-body"
            >
              <p
                class="MuiTypography-root makeStyles-largeText-2 MuiTypography-body1"
              >
                <span>
                  baz
                </span>
                 
              </p>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root makeStyles-cell-1 MuiTableCell-body"
            >
              <p
                class="MuiTypography-root makeStyles-largeText-2 MuiTypography-body1"
              >
                <span>
                  quox
                </span>
                 
              </p>
            </td>
            <td
              class="MuiTableCell-root makeStyles-cell-1 MuiTableCell-body"
            >
              <p
                class="MuiTypography-root makeStyles-largeText-2 MuiTypography-body1"
              >
                <span>
                  apple
                </span>
                 
              </p>
            </td>
            <td
              class="MuiTableCell-root makeStyles-cell-1 MuiTableCell-body"
            >
              <p
                class="MuiTypography-root makeStyles-largeText-2 MuiTypography-body1"
              >
                <span>
                  banana
                </span>
                 
              </p>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root makeStyles-cell-1 MuiTableCell-body"
            >
              <p
                class="MuiTypography-root makeStyles-largeText-2 MuiTypography-body1"
              >
                <span>
                  chad
                </span>
                 
              </p>
            </td>
            <td
              class="MuiTableCell-root makeStyles-cell-1 MuiTableCell-body"
            >
              <p
                class="MuiTypography-root makeStyles-largeText-2 MuiTypography-body1"
              >
                <span>
                  elle
                </span>
                 
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
